cucumber-core/src/main/java/io/cucumber/core/eventbus/FastUuidGenerator.java

```diff
@@ -0,0 +1,32 @@
+package io.cucumber.core.eventbus;
+
+import io.cucumber.core.exception.CucumberException;
+
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * Thread-safe UUID generator for single JVM. This is a sequence generator and
+ * each instance has its own counter. This generator is about 100 times faster
+ * than #DefaultUuidGenerator. If you use Cucumber in multi-JVM setup, you
+ * should use #DefaultUuidGenerator instead.
+ */
+public class FastUuidGenerator implements UuidGenerator {
+    private final AtomicLong counter = new AtomicLong(Long.MIN_VALUE);
+
+    /**
+     * Generate a new UUID. Will throw an exception when out of capacity.
+     * 
+     * @return                   a non-null UUID
+     * @throws CucumberException when out of capacity
+     */
+    @Override
+    public UUID get() {
+        long leastSigBits = counter.incrementAndGet();
+        if (leastSigBits == Long.MAX_VALUE) {
+            throw new CucumberException(
+                "Out of FastUuidGenerator capacity. Please use the DefaultUuidGenerator instead.");
+        }
+        return new UUID(Thread.currentThread().getId(), leastSigBits);
```

Comment:
What is the purpose of the current thread ID? I reckon we are tempting Hyrum's Law by using it.

If a value must be used here, we could use a random value that is determined on initialization (a session Id of sorts). That would solve some of the collision concerns.

Reply:
True, I made a mistake here, mixing the "thread-safety" with the "collision" features. The thread-safe feature is provided through the `AtomicLong`, so we don't need the current thread identifier to at the generation time. 

To avoid the collisions, using the current thread identifier may not be the best approach, because one thread could have two `IncrementingUuidGenerator`s, When using a pure random session identifier is not ideal as well, as there is a small collision risk. But an incrementing session identifier (e.g. `static final AtomicLong` field) would do the trick. I'll refactor.